projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/service/KubernetesServiceTest.java

```diff
@@ -218,4 +218,45 @@ public void testCreateV1beta1CronJobFromInternalResource() {
         }
     }
 
+    @Test
+    public void testQuantityToMbConversionMegabytes() throws Exception {
+        var hundredMb = "100M";
+        testQuantityToMbConversion(100, hundredMb);
+
+        var thousandMb = "1000M";
+        testQuantityToMbConversion(1000, thousandMb);
+        // this should be enough for overflow errors on the method to manifest.
+        var tenThousandMb = "10000M";
+        testQuantityToMbConversion(10000, tenThousandMb);
+
+        var hundredThousandMb = "100000M";
+        testQuantityToMbConversion(100000, hundredThousandMb);
+
+    }
+
+    @Test
+    public void testQuantityConversionGigabytes() throws Exception {
+        var oneG = "1G";
+        testQuantityToMbConversion(1000, oneG);
+
+        var twoG = "2G";
+        testQuantityToMbConversion(2000, twoG);
+        // this should be enough for overflow errors on the method to manifest.
+        var threeG = "3G";
+        testQuantityToMbConversion(3000, threeG);
+
+        var fourG = "4G";
+        testQuantityToMbConversion(4000, fourG);
+
+        var sixtyFourG = "64G";
+        testQuantityToMbConversion(64000, sixtyFourG);
+    }
+
+    public void testQuantityToMbConversion(int expectedMb, String providedResources) throws Exception {
+        KubernetesService service = new DataJobsKubernetesService("default", "someConfig");
```

Comment:
Since convertMemoryToMBs() is static it should be available at class level - KubernetesService.convertMemoryToMBs(q);